client/src/services/recipe/recipe.service.ts

```diff
@@ -29,9 +41,7 @@ export class RecipeService {
     return this.httpClientService.get<Recipe>(`recipe/${id}`);
   }
 
-  public searchRecipes(searchQuery: string): Observable<Recipe[]> {
-    return this.httpClientService.post<Recipe[]>('recipe/search', {
-      searchQuery: searchQuery
-    });
+  public searchRecipes(searchModel: SearchModel): Observable<Recipe[]> {
+    return this.httpClientService.post<Recipe[]>('recipe/search', searchModel);
```

Comment:
It would be better to generate the model here instead as it directly ties to the backend model. Having it here ensures we only have to refactor this if needed.